_source/_posts/2024-07-05-jhipster-terraform-azure.md

```diff
@@ -0,0 +1,1857 @@
+  name                  = "dns-link-acr"
+  resource_group_name   = azurerm_resource_group.rg_spoke_networks.name
+  private_dns_zone_name = azurerm_private_dns_zone.dns_zone_acr.name
+  virtual_network_id    = azurerm_virtual_network.spoke_vnet.id
+}
+
+resource "azurerm_public_ip" "spoke_pip" {
+  name                = local.pip_name
+  location            = azurerm_resource_group.rg_spoke_networks.location
+  resource_group_name = azurerm_resource_group.rg_spoke_networks.name
+  allocation_method   = "Static"
+  sku                 = "Standard"
+  zones              = ["1","2", "3"]
+  idle_timeout_in_minutes = 4
+  ip_version = "IPv4"
+}
+
+resource "azurerm_application_gateway" "gateway" {
+  name                = "app-gateway"
+  location            = azurerm_resource_group.rg_spoke_networks.location
+  resource_group_name = azurerm_resource_group.rg_spoke_networks.name
+  zones               = ["1", "2", "3"]
+
+  sku {
+    name     = "WAF_v2"
+    tier     = "WAF_v2"
+    capacity = 2
+  }
+
+  gateway_ip_configuration {
+    name      = "gateway-ip-configuration"
+    subnet_id = azurerm_subnet.application_gateways_subnet.id
+  }
+
+  frontend_port {
+    name = local.frontend_port_name
+    port = 80
+  }
+
+  frontend_ip_configuration {
+    name                 = local.frontend_ip_configuration_name
+    public_ip_address_id = azurerm_public_ip.spoke_pip.id
+  }
+
+  waf_configuration {
+    enabled = true
+    firewall_mode = "Prevention"
+    rule_set_type = "OWASP"
+    rule_set_version = "3.0"
+  }
+
+  backend_address_pool {
+    name = local.backend_address_pool_name
+  }
+
+  backend_http_settings {
+    name                  = local.http_setting_name
+    cookie_based_affinity = "Disabled"
+    path                  = "/path1/"
+    port                  = 80
+    protocol              = "Http"
+    pick_host_name_from_backend_address = true
+    request_timeout       = 60
+  }
+
+  http_listener {
+    name                           = local.listener_name
+    frontend_ip_configuration_name = local.frontend_ip_configuration_name
+    frontend_port_name             = local.frontend_port_name
+    protocol                       = "Http"
+    host_name                      = var.host_name
+    #require_sni                    = true
+  }
+
+  request_routing_rule {
+    name                       = local.request_routing_rule_name
+    priority                   = 1
+    rule_type                  = "Basic"
+    http_listener_name         = local.listener_name
+    backend_address_pool_name  = local.backend_address_pool_name
+    backend_http_settings_name = local.http_setting_name
+  }
+}
+```
+The configuration above will create a Spoke Network, the network peerings the hub and spoke networks, and the Azure application gateway hosted in a dedicated subnet.
+
+Edit `outputs.tf` and add the following content:
+
+```terraform
+# terraform/modules/spoke_network/outputs.tf
```

Comment:
Lets also remove outputs that are not useful for any further usecase

Reply:
Removed unused outputs.